internal/util/util.go

```diff
@@ -392,13 +393,14 @@ func MergeMaps(maps ...map[string]string) map[string]string {
 	return out
 }
 
-func LoadCertPool(certFile string) (*x509.CertPool, error) {
-	rootCAPEM, err := os.ReadFile(certFile)
+func LoadCertPool(rootCAPEM io.Reader) (*x509.CertPool, error) {
```

Comment:
Adding a godoc here would be a good follow-up imo.

Reply:
I added a short comment.